src/overlays/actors/ovl_En_Tg/z_en_tg.c

```diff
@@ -50,23 +39,154 @@ static ColliderCylinderInit D_80B18910 = {
+    EnTg* this = THIS;
+    u16 temp;
+    u32 phi;
+
+    temp = Text_GetFaceReaction(globalCtx, 0x24);
+    if (temp != 0) {
+        return temp;
+    }
+    if (globalCtx->sceneNum == 0x52) {
+        if (this->unk_208 & 1) {
+            phi = 0x5089;
+        } else {
+            phi = 0x508A;
+        }
+        return phi;
+    } else {
+        if (this->unk_208 & 1) {
+            phi = 0x7025;
+        } else {
+            phi = 0x7026;
+        }
+        return phi;
+    }
+}
+
+s16 func_80B183F8(GlobalContext* globalCtx, Actor* thisx) {
+    EnTg* this = THIS;
+
+    switch (func_8010BDBC(&globalCtx->msgCtx)) {
+        case 0:
+        case 1:
+        case 3:
+        case 4:
+        case 5:
+        case 6:
+        case 7:
+        case 8:
+        case 9:
+            return 1;
+        case 2:
+            switch (this->actor.textId) {
+                case 0x5089:
+                case 0x508A:
+                    this->unk_208++;
+                    break;
+                case 0x7025:
+                case 0x7026:
+                    this->actor.params ^= 1;
+                    this->unk_208++;
+                    break;
+            }
+            return 0;
+        default:
+            return 1;
+    }
+}
+
+void EnTg_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnTg* this = THIS;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 28.0f);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &D_0600AE40, &D_06005040, NULL, NULL, 0);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, NULL, &sColChkInfoInit);
+    this->actor.targetMode = 6;
+    Actor_SetScale(&this->actor, 0.01f);
+    this->unk_208 = globalCtx->state.frames % 2;
+    this->actionFunc = EnTg_SpinIfNotTalking;
+}
+
+void EnTg_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnTg* this = THIS;
+
+    SkelAnime_Free(&this->skelAnime, globalCtx);
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnTg_SpinIfNotTalking(EnTg* this, GlobalContext* globalCtx) {
+    if (this->isTalking == 0) {
+        this->actor.shape.rot.y += 0x800;
+    }
+}
+
+void EnTg_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnTg* this = THIS;
+    s32 pad;
+    f32 temp;
+    Vec3s sp2C;
+
+    sp2C.x = this->actor.world.pos.x;
+    sp2C.y = this->actor.world.pos.y;
+    sp2C.z = (s16)this->actor.world.pos.z + 3;
+    this->collider.dim.pos = sp2C;
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    SkelAnime_Update(&this->skelAnime);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 0.0f, 0.0f, 0.0f, 4);
+    this->actionFunc(this, globalCtx);
+    temp = this->collider.dim.radius + 30.0f;
+    func_800343CC(globalCtx, &this->actor, &this->isTalking, temp, func_80B18360, func_80B183F8);
```

Comment:
Would this match?
```suggestion
    func_800343CC(globalCtx, &this->actor, &this->isTalking, this->collider.dim.radius + 30.0f, func_80B18360, func_80B183F8);
```

Reply:
If I remember correctly from last night, it does not.